lumispy/utils/analysis.py

```diff
@@ -0,0 +1,117 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019-2022 The LumiSpy developers
+#
+# This file is part of LumiSpy.
+#
+# LumiSpy is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the license, or
+# (at your option) any later version.
+#
+# LumiSpy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with LumiSpy. If not, see <https://www.gnu.org/licenses/#GPL>.
+
+import hyperspy.api as hs
+
+
+def plot_span_map(sig, nspans=1):
+    """
+    Plot a span map.
+    
+    Plots a navigator consisting of the sum over all positions in sig, with `nspans` spans. 
+
+    For each span a `BaseSignal` image is plotted consisting of integrated counts over the the range defined by the span.
+
+    The spans can be moved interactively and the corresponding images will update automatically.
+
+    Arguments
+    ---------
+    sig: hyperspy.signals.Signal1D
+        hyperspectra to inspect. Should have 2 spatial dimensions and 1 signal dimension.
+    nspans: int
+        number of channels to plot. Maximum is 3, default is 1.
+
+    Returns
+    -------
+    all_sum: hs.signals.BaseSignal
+        Sum over all positions of `sig`, the 'navigator' for `plot_span_map`
+    spans: hs.roi.SpanROI
+        The span ROIs from the navigator
+    span_sigs: hs.signals.BaseSignal
+        slices of `sig` according to each span roi
+    span_sums: hs.signals.BaseSignal
+        the summed `span_sigs`.
+    """
+    if nspans > 3:
+        raise ValueError("Maximum number of spans allowed is 3")
+
+    if (
+        sig.axes_manager.signal_dimension != 1
+        or sig.axes_manager.navigation_dimension != 2
+    ):
+        sig_dims, nav_dims = (
+            sig.axes_manager.signal_dimension,
+            sig.axes_manager.navigation_dimension,
+        )
+        raise ValueError(
+            f"This method is designed for data with 1 signal and 2 navigation dimensions, not {sig_dims} and {nav_dims} respectively"
+        )
+
+    colors = ["red", "green", "blue"]
```

Comment:
Could you have this as an argument too?  Happy to have these 3 as default, but it would be nice to be able to change that.

Reply:
Discussed above.